vec3_msg[] coords
---
map_block_msg[] blocks
